quodlibet/ext/events/advanced_preferences.py

```diff
@@ -218,7 +218,14 @@ def changed(entry, name, section="settings"):
                 "Minimum length proportion to consider a track as played:",
                 ("Consider a track played after listening to this proportion of "
                  "its total duration"),
-                label_value_callback=lambda value: f"{int(value * 100)}%")
+                label_value_callback=lambda value: f"{int(value * 100)}%"),
+            text_config(
+                "browsers", "missing_title_template",
+                "Missing title template string:",
+                ("Template for building title of tracks when title tag is missing. "
+                 "{basename} is expanded to the filename of the track, "
```

Comment:
It might be nicer / "more QL" to use a QL `Pattern` for this, like many other things do (Album Browser, CoverGrid, Playlist), file renames, lots of plugins: scrobbler, Website Search, Gajim, MediaServer, etc) .

Then users can specify `<~basename> <~dirname> <~format> <~length> <~#bitrate> <~filesize>` and more, including any tags it _might_ have and any default logic they want. :up:

Reply:
Good, I will check that and be back. Thanks.

Reply:
Hi,
Some changes you can check.
- using Pattern
- 2 new tags: <~basename:stem> and <~basename:humanized>

plz check if it's implemented in the right way... I don't have 360° eyes :)

I will update the test/doc later if it's ok.

Read you, J.

Reply:
Yes this looks much better, thanks.

I **don't** think we should add any new _tags_  / keys in this PR though (and thus think `<~basename>` is probably fine for now, as it also mimics existing functionality). All keys need documenting and sometimes there are considerations in other places around the code (e.g. proxies, aggregations), that are far off the  scope of #3872.

Also this string / tooltip should be updated to talk about normal patterns (there are a few others around if you want to use the same style, for translation ease etc)